### DISCUSSION TITLE
How does quadratic sampling work?

### DISCUSSION THREAD

**Discussion by Azirine** [LostRuins/koboldcpp#738] [Q&A]
It is not clear how "we transform each logit based on a quadratic function with a scaling factor & a reference value (h)". What are the exact equations used? Kalomaze appears to have an interactive visualizer but we only get an image of it and a video of him using it. It would be useful if we could use it too.
![299838025-53046e7f-5909-492c-a52a-2d9e365be917](https://github.com/LostRuins/koboldcpp/assets/101342773/0185c688-f5f7-4e38-9a8c-ec7af133a536)
https://files.catbox.moe/x804ia.mp4

**Comment by LostRuins** [comment]
I cannot advise on this sampler as I personally don't understand how it works - the implementation was just added wholesale. You would have to ask @kalomaze.
